library/src/main/java/com/novoda/downloadmanager/BatchStorageRoot.java

```diff
@@ -0,0 +1,21 @@
+package com.novoda.downloadmanager;
+
+import java.io.File;
+
+final class BatchStorageRoot {
```

Comment:
I don't understand the necessity of adding this. The `StorageRoot` and the `BatchId` give you everything you need in order to be able to delete a `folder`. This feels like it is doing the same thing as the `Builder` is already doing but just wrapping it again into another class. ⚾️

Reply:
Ok. So this exists to keep the `storage root` and `downloadBatchId` tied together. Fair enough.

Reply:
Should this be auto-valued given it is a model class?

Reply:
I think we didn't add AutoValue in this project no?

Reply:
Oh 🤔 probably not actually. Well in that case, `equals` and `hashcode` and `toString` 😂

Reply:
Added those three methods.